### PR TITLE
Add Eigen Quaterniond convenience method.

### LINKED ISSUE #35: add .getRotationQuaternion() to minkindr transform?
```
Eigen::Quaterniond q_odom_map = T_odom_map_.getRotation().toImplementation();
```
is a hassle every time you need the rotation component of this transformation, which is all the time.

Any objections to adding a convenience method?

### PR DESCRIPTION
**Pull request by helenol**
Currently only the const version is implemented: any opinions on this? I am personally happy with just const, and it feels safer this way (just a getter, not a setter).

@schneith @dymczykm 

Resolves #35

### CODE REVIEW

**Comment by helenol** [comment]
test this please (removed broken dep on a weird branch of eigen_catkin and dep on gtsam_catkin)

File: minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h [hunk 1]
```diff
@@ -116,6 +116,12 @@ QuatTransformationTemplate<Scalar>::getRotation() const {
   return q_A_B_;
 }
 
+template <typename Scalar>
+const Eigen::Quaterniond&
```
Comment (dymczykm) [inline minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h]: shouldn't this be templated on the scalar type??

**Comment by dymczykm** [comment]
one q, olgtm

File: minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h (hunk 1 above)
Comment (helenol) [inline minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h]: That does, uh, make an awful lot of sense. Thanks for catching that, fixing.

**Pull request merged**